pandas/core/tools/datetimes.py

```diff
@@ -275,9 +275,14 @@ def _convert_listlike_datetimes(arg, box, format, name=None, tz=None,
                 yearfirst=yearfirst,
                 require_iso8601=require_iso8601
             )
-            if tz_parsed is not None and box:
-                return DatetimeIndex._simple_new(result, name=name,
-                                                 tz=tz_parsed)
+            if tz_parsed is not None:
+                if box:
```

Comment:
can you add some comments here would be great (just to make it easier to read, the block below as well)